lua/telescope/builtin.lua

```diff
@@ -950,4 +950,79 @@ builtin.keymaps = function(opts)
   }):find()
 end
 
+builtin.tags = function(opts)
+  opts = opts or {}
+
+  if not vim.loop.fs_open('tags', "r", 438) then
+    print('Tags file does not exists. Create one with ctags -R')
+    return
+  end
+
+  local fd = assert(vim.loop.fs_open('tags', "r", 438))
+  local stat = assert(vim.loop.fs_fstat(fd))
+  local data = assert(vim.loop.fs_read(fd, stat.size, 0))
+  assert(vim.loop.fs_close(fd))
+
+  local lines = vim.split(data, '\n')
+
+  pickers.new(opts,{
+    prompt = 'Tags',
+    finder = finders.new_table {
+      results = lines,
+      entry_maker = make_entry.gen_from_ctags(opts),
+    },
+    previewer = previewers.ctags.new(opts),
+    sorter = conf.generic_sorter(opts),
+    attach_mappings = function(prompt_bufnr, map)
+      local update_lnum = actions._transform_action(function(prompt_bufnr)
+        local selection = actions.get_selected_entry(prompt_bufnr)
+        local strip_endline = string.gsub(selection.scode, '[$]$', '')
+        local strip_doubleslash = string.gsub(strip_endline, [[\\]], [[\]])
+        local strip_c_comment = string.gsub(strip_doubleslash, [[\/]], [[/]])
+
+        local cmd = 'grep -Frn \'' .. strip_c_comment .. '\' ' .. selection.filename
+        local output = vim.split(utils.get_os_command_output(cmd), ':')
+
+        selection.lnum = tonumber(output[1])
+      end)
+
+      map('i', '<CR>', update_lnum + actions.goto_file_selection_edit)
+      map('n', '<CR>', update_lnum + actions.goto_file_selection_edit)
+      map('i', '<C-x>', update_lnum + actions.goto_file_selection_split)
+      map('n', '<C-x>', update_lnum + actions.goto_file_selection_split)
+      map('i', '<C-v>', update_lnum + actions.goto_file_selection_vsplit)
+      map('n', '<C-v>', update_lnum + actions.goto_file_selection_vsplit)
+      map('i', '<C-t>', update_lnum + actions.goto_file_selection_tabedit)
+      map('n', '<C-t>', update_lnum + actions.goto_file_selection_tabedit)
+      return true
+    end
+    -- TODO(conni2461): I kinda want something like this, to respect user mappings
+    -- inject_action = function(prompt_bufnr, inject)
+    --   local update_lnum = actions._transform_action(function(prompt_bufnr)
+    --     local selection = actions.get_selected_entry(prompt_bufnr)
+    --     local strip_endline = string.gsub(selection.scode, '[$]$', '')
+    --     local strip_doubleslash = string.gsub(strip_endline, [[\\]], [[\]])
+    --     local strip_c_comment = string.gsub(strip_doubleslash, [[\/]], [[/]])
+
+    --     local cmd = 'grep -Frn \'' .. strip_c_comment .. '\' ' .. selection.filename
+    --     local output = vim.split(utils.get_os_command_output(cmd), ':')
+
+    --     selection.lnum = tonumber(output[1])
+    --   end)
+    --   inject({
+    --     actions.goto_file_selection_edit,
+    --     actions.goto_file_selection_split,
+    --     actions.goto_file_selection_vsplit,
+    --     actions.goto_file_selection_tabedit},
+    --     { before = update_lnum }
+    --   )
+    -- end
```

Comment:
@tjdevries Does this make sense. Rather than using attach_mappings, so we can respect user_mappings?

Reply:
oh I see, the idea is basically to evolve certain actions into doing something special for this case?

So they map to `X` but we do `f(X)` or something.

Reply:
@tjdevries So basically they have changed `split` to `[C-w]` or something and that will run `actions.goto_file_selection_split`. But if i have a new action now that has to run before, after or completely replace that one, i currently use `attach_mappings` and map this `[C-x]` because this is default. But if the user uses now his mapping it will not run my new action.

I think i need an example. For tags i have to open up the file with `actions.goto_file_selection_` and than i have to do the search. So i can do `actions.goto_file_selection_ + jump_to_tag`. And now i don't want this to happen only on that attached mapping but on any keymapping that runs `actions.goto_file_selection_`. Does this make sense?

Btw i ran into a problem when doing `actions.goto_file_selection_ + jump_to_line`. When my function would run i no longer can get the selection with `actions.get_selected_entry(prompt_bufnr)` because goto_file_selection already closes the picker. Maybe we should offer a lightway version of `goto_file_selection_` which does not close the picker.